boa_engine/src/context/mod.rs

```diff
@@ -101,6 +102,9 @@ pub struct Context<'host> {
     host_hooks: &'host dyn HostHooks,
 
     job_queue: &'host dyn JobQueue,
+
+    optimize: bool,
```

Comment:
Should we look at narrowing this down to a bitflag eventually?

Reply:
Maybe, I also though of a having a struct `OptimizerOptions` where we can enable or disable certain optimizations (should make much easier to debug optimizations and see how they perform, since they can interfere with each other, that why they are run one after the other)

Maybe have a CLI flag comma separated list `--optimize=constant-folding,constant-propagation,dead-code-elimination` by default enabling all, if nothing specified

Or maybe I should leave this for when we have more than one optimization? :sweat_smile: 

What do you think about this? @nekevss

Reply:
> Should we look at narrowing this down to a bitflag eventually?

I put it in bitflag in the latest commit :)

Reply:
I like it! There's probably more things we could consider in the future like a `default` method that sets all opt-out flags. But overall, this looks good to me 😄